src/construction.cpp

```diff
@@ -851,16 +881,29 @@ void complete_construction()
 
         award_xp( *elem );
     }
-
-    for( const auto &it : built.requirements->get_components() ) {
-        u.consume_items( it, 1, is_crafting_component );
+    // Remove the unfinished item marker
+    item *con_item = u.activity.targets.front().get_item();
```

Comment:
How about just calling `item_location::remove_item()`?
```suggestion
    u.activity.targets.front().remove_item();
```